rust/src/web/graphql/accounts/mod.rs

```diff
@@ -110,15 +108,71 @@ impl AccountQueryRoot {
     }
 }
 
+impl AccountQueryInput {
+    fn matches(&self, account: &account::Account) -> bool {
+        let AccountQueryInput {
+            public_key,
+            // username,
+            balance,
+            balance_gt,
+            balance_gte,
+            balance_lt,
+            balance_lte,
+            balance_ne,
+        } = self;
+        if let Some(public_key) = public_key {
+            return *public_key == account.public_key.0;
+        }
+        // if let Some(username) = username {
+        //     return account
+        //         .username
+        //         .as_ref()
+        //         .map_or(false, |u| *username == u.0);
+        // }
```

Comment:
Leave commented out or remove??

Reply:
I'll leave the comment until we handle usernames